nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/StandardProtobufReader.java

```diff
@@ -0,0 +1,306 @@
+            final SchemaDefinition schemaDefinition = schemaRegistry.retrieveSchemaDefinition(schemaIdentifier);
+            logger.debug("Using message name for schema identifier: {}", schemaDefinition.getIdentifier());
+            return createProtobufRecordReader(variables, in, schemaDefinition);
+        }
+
+        throw new SchemaNotFoundException("Unsupported schema access strategy: " + schemaAccessStrategyValue);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>(super.getSupportedPropertyDescriptors());
+        properties.add(MESSAGE_NAME_RESOLVER_STRATEGY);
+        properties.add(MESSAGE_NAME_RESOLVER_CONTROLLER_SERVICE);
+        properties.add(MESSAGE_NAME);
+        return properties;
+    }
+
+    @Override
+    protected PropertyDescriptor buildSchemaTextProperty() {
+        return new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(SCHEMA_TEXT)
+            .required(true)
+            .clearValidators()
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .defaultValue("${proto.schema}")
+            .description("The text of a Proto 3 formatted Schema")
+            .build();
+    }
+
+    private RecordReader createProtobufRecordReader(final Map<String, String> variables, final InputStream in, final SchemaDefinition schemaDefinition) throws IOException {
+        final Schema schema = schemaCompiler.compileOrGetFromCache(schemaDefinition);
+        final ProtoSchemaParser schemaParser = new ProtoSchemaParser(schema);
+        final MessageName messageName = messageNameResolver.getMessageName(variables, schemaDefinition, in);
+        final RecordSchema recordSchema = schemaParser.createSchema(messageName.getFullyQualifiedName());
+        return new ProtobufRecordReader(schema, messageName.getFullyQualifiedName(), in, recordSchema);
+    }
+
+
+    private void setupMessageNameResolver(final ConfigurationContext context) {
+        final MessageNameResolverStrategyName messageNameResolverStrategyName = context.getProperty(MESSAGE_NAME_RESOLVER_STRATEGY).asAllowableValue(MessageNameResolverStrategyName.class);
+        messageNameResolver = switch (messageNameResolverStrategyName) {
+            case MESSAGE_NAME_PROPERTY -> new PropertyMessageNameResolver(context);
+            case MESSAGE_NAME_RESOLVER_SERVICE -> context.getProperty(MESSAGE_NAME_RESOLVER_CONTROLLER_SERVICE).asControllerService(MessageNameResolver.class);
+        };
+    }
+
+    private SchemaDefinition createSchemaDefinitionFromText(final Map<String, String> variables) throws SchemaNotFoundException {
+        final String schemaTextString = schemaText.evaluateAttributeExpressions(variables).getValue();
+        validateSchemaText(schemaTextString);
+
+        final String sha256hex = sha256Hex(schemaTextString);
```

Comment:
Minor, but recommend renaming this since this implementation details of the hash are less important:
```suggestion
        final String hash = getHash(schemaTextString);
```

Reply:
Done